### PR TITLE
group posts by year

### PR DESCRIPTION
**Pull request by Tevinthuku**
Filter out the blog posts by the dates.

image

### CODE REVIEW

File: src/pages/blog/typography.md [hunk 1]
```diff
@@ -75,18 +75,18 @@ But let's throw in a <b>tag</b>.
...
+| col 2 is      |   centered    |   $12 |
+| zebra stripes |   are neat    |    $1 |
 
 ### Table 2: With Typography Elements
 
-Markdown | Less | Pretty
---- | --- | ---
-*Still* | `renders` | **nicely**
-1 | 2 | 3
+| Markdown | Less      | Pretty     |
```
Comment (wangonya) [inline src/pages/blog/typography.md:86]: I know prettier or eslint probably did this but the markdown table here wasn't meant to be spaced out correctly. The point was to show that the table **still renders nicely** even if the spacing is off. I'll revert to the original.

File: src/pages/blog/typography.md [hunk 2]
```diff
@@ -116,7 +116,7 @@ ___
 
 ### Image
 
-![Minion](http://octodex.github.com/images/minion.png)
+![Minion](https://octodex.github.com/images/minion.png)
```
Comment (wangonya) [inline src/pages/blog/typography.md:119]: Thanks for this! The Netlify build kept warning me about it but I was too lazy to fix it 😅

File: src/pages/index.js [hunk 3]
```diff
@@ -5,27 +5,39 @@ import DefaultLayout from "../layouts/default"
...
+        <div className="title">
+          <Link to={post.node.fields.slug}>{post.node.frontmatter.title}</Link>
+        </div>
+      </li>
+    ))
+
+  const postsListContainer = groupBy(posts, getDateYear).map(
+    ({ year, posts }, i) => (
+      <div key={i}>
+        <b>{year}</b>
```
Comment (wangonya) [inline src/pages/index.js:31]: All it needs is some styling and we'll be good to 🚀. Let me work on it. Thanks!

File: src/pages/blog/typography.md (hunk 2 above)
Comment (Tevinthuku) [reply src/pages/blog/typography.md:119]: 😆yeap that was the only thing.

File: src/pages/index.js (hunk 3 above)
Comment (Tevinthuku) [reply src/pages/index.js:31]: Awesome 👍

**Pull request merged**